tests/shared/src/test/scala/scala/meta/tests/parsers/ScaladocParserSuite.scala

```diff
@@ -0,0 +1,425 @@
+    val complexCodeBlockAsComment = complexCodeBlock.mkString("\n *")
+
+    val result =
+      parseString(
+        s"""
+          /**
+            * $testDescription {{{ $codeBlock1 }}}
+            * $testDescription
+            * {{{ $codeBlock2 }}}
+            *
+            * $testDescription
+            *
+            * {{{
+            *$complexCodeBlockAsComment
+            * }}}
+            * {{{
+            *$complexCodeBlockAsComment }}}
+            */
+       """.stripMargin
+      )
+
+    val expectation = Option(
+      Scaladoc(
+        Paragraph(Text((words :+ CodeExpr(codeBlock1)) ++ (words :+ CodeExpr(codeBlock2)): _*)),
+        Paragraph(Text(words: _*)),
+        Paragraph(CodeBlock(complexCodeBlock: _*), CodeBlock(complexCodeBlock: _*))
+      )
+    )
+    assertEquals(result, expectation)
+  }
+
+  test("headings") {
+    val level1HeadingBody = "Level 1"
+    val level2HeadingBody = "Level 2"
+    val level3HeadingBody = "Level 3"
+    val level4HeadingBody = "Level 4"
+    val level5HeadingBody = "Level 5"
+    val level6HeadingBody = "Level 6"
+    val bad1 = "=======7isTooMuch======="
+    val bad2 = "======6left,5right====="
+    val bad3 = "=====5left,5right=====debris"
+    val bad4 = "debris=====5left,5right====="
+
+    val result =
+      parseString(
+        s"""
+        /**
+          * =$level1HeadingBody=
+          * ==$level2HeadingBody==
+          * ===$level3HeadingBody===
+          * ====$level4HeadingBody====
+          * =====$level5HeadingBody=====
+          * ======$level6HeadingBody======
+          * $bad1
+          * $bad2
+          * $bad3
+          * $bad4
+          */
+         """
+      )
+    val expectation = Option(
+      Scaladoc(
+        Paragraph(
+          Heading(1, level1HeadingBody),
+          Heading(2, level2HeadingBody),
+          Heading(3, level3HeadingBody),
+          Heading(4, level4HeadingBody),
+          Heading(5, level5HeadingBody),
+          Heading(6, level6HeadingBody),
+          Text(Word(bad1)),
+          Text(Word(bad2)),
+          Text(Word(bad3), Word(bad4))
+        )
+      )
+    )
+    assertEquals(result, expectation)
+  }
+
+  test("lists") {
+    val list11 = "List11"
+    val list12 = "List12"
+    val list21 = "List21"
+    val list22 = "List22"
+    val list31 = "List31"
+    val list32 = "List32"
+    val list41 = "List41"
+    val list42 = "List42"
+
+    val result =
+      parseString(
+        s"""
+        /**
+          * Some text:
+          * - $list11
+          *    1. $list21
+          *    1. $list22
+          *       i. $list31
+          *        a. $list41
+          *        a. $list42
+          *       i. $list32
+          * - $list12
+          *    - $list21
+          *       I. $list31
+          *        - $list41
+          *        - $list42
+          *       I. $list32
+          *    - $list22
+          */
+         """
+      )
+    val expected = Option(
```

Comment:
This value is a bit too big to produce helpful diffs on test failures. I would consider using a smaller example

Reply:
done. split into several tests.